cmd/network/netintf/netintf.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (c) 2025 Dell Inc, or its subsidiaries.
+
+// Package netintf implements the network interface related CLI commands
+package netintf
+
+import (
+	"context"
+	"log"
+
+	"github.com/opiproject/godpu/cmd/common"
+	network "github.com/opiproject/godpu/network"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+// ListNetworkInterfaces lists all Network Interface details from OPI server
+func ListNetworkInterfaces() *cobra.Command {
+	var pageSize int32
+	var pageToken string
+
+	cmd := &cobra.Command{
+		Use:   "list-net-interfaces",
+		Short: "List the network interfaces",
+		Run: func(c *cobra.Command, _ []string) {
+			addr, err := c.Flags().GetString(common.AddrCmdLineArg)
+			cobra.CheckErr(err)
+
+			timeout, err := c.Flags().GetDuration(common.TimeoutCmdLineArg)
+			cobra.CheckErr(err)
+
+			tlsFiles, err := c.Flags().GetString(common.TLSFiles)
+			cobra.CheckErr(err)
+
+			ctx, cancel := context.WithTimeout(context.Background(), timeout)
+
+			netifClient, err := network.NewNetInterface(addr, tlsFiles)
+			if err != nil {
+				log.Fatalf("could not create gRPC client: %v", err)
+			}
+			defer cancel()
```

Comment:
would it make sense to create `ctx` and `cancel` right here and keep them close to this defer call?

Reply:
Will look at this one to determine if it needs to move or change in future addition.